### DISCUSSION TITLE
Scribe SLF4J2 and doobie-hikari issues

### DISCUSSION THREAD

**Discussion by MirkoManojlovic** [outr/scribe#405] [Q&A] [answered]
Hello, 

I'm creating a standard stack backend app using htt4s, quill-doobie and of course scribe :)

App was working fine (in a sense that logging was working), but since I've introduced quill-doobie and doobie-hikari lib, I started getting this on the app startup

```bash
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.
SLF4J: Class path contains SLF4J bindings targeting slf4j-api versions prior to 1.8.
SLF4J: Ignoring binding found at [jar:file:/Users/mirko/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/com/outr/scribe-slf4j_2.13/3.11.0/scribe-slf4j_2.13-3.11.0.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#ignoredBindings for an explanation.
```

So I went to the link https://www.slf4j.org/codes.html#ignoredBindings, and see that some versions are off. Slf4j version 1.8 and higher use ServiceLoader mechanism instead of StaticBinding. I saw in the code that there is some work on slf4j2 version https://github.com/outr/scribe/blob/master/build.sbt#L57, and I also took a look at the the doobie lib. The doobie folks seems to have the same issue.

https://github.com/tpolecat/doobie/blob/main/build.sbt#L10

https://github.com/tpolecat/doobie/blob/main/build.sbt#L329-L330

I've tried to exclude myself, but it's still not working. I was wondering is the slf4j2 version ready? Or something else I'm missing?
And also sorry if this is off topic (not related to Scribe)

**Comment by darkfrog26** [comment]
The reason that both Scribe and Doobie don't rely on 1.8 is that it's still in beta. The `slf4j2` binding should work fine if you're able to upgrade. Let me know if that solves your problem or if there's something else I can do. I've seen this in the past be resolved by changing the order of your dependencies. Sometimes just swapping the order in which they are listed can sometimes fix it.

**Comment by MirkoManojlovic** [comment, marked as answer]
Hmm yeah I tried the dependency order swap, didn't work. Anyways it appears I just suck at SBT 😄 

Instead of excluding it I should have been "forcing it". When I added this line to project settings

`dependencyOverrides += "org.slf4j" % "slf4j-api" % "1.7.36",`

it started working again. Something about `quill-doobie` and `doobie-hikari` libs was off. But anyway sorry for the trouble, thank you for the answer and for the time and for the awesome lib 😃

**Comment by darkfrog26** [comment]
No worries, glad you got it figured out!
